src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,83 @@
-export const Pagination = () => {};
+import classNames from 'classnames';
+import React from 'react';
+
+interface PaginationProps {
+  total: number;
+  perPage: number;
+  currentPage?: number;
+  onPageChange: (page: number) => void;
+}
+
+export const Pagination: React.FC<PaginationProps> = ({
+  total,
+  perPage,
+  currentPage = 1,
+  onPageChange,
+}) => {
+  const totalPages = Math.ceil(total / perPage);
+
+  const handlePageChange = (page: number) => {
+    if (page !== currentPage && page >= 1 && page <= totalPages) {
+      onPageChange(page);
+    }
+  };
+
+  return (
+    <ul className="pagination">
+      <li className={classNames('page-item', { disabled: currentPage === 1 })}>
+        <a
+          data-cy="prevLink"
+          className="page-link"
+          href="#prev"
+          aria-disabled={currentPage === 1}
```

Comment:
Using `href="#prev"` and `href="#next"` for the previous and next page links can lead to potential accessibility issues and unexpected behavior when users click on them. Instead of using anchor tags with `href` attributes, consider using button elements which are more appropriate for actions that do not navigate to a different URL. This would also eliminate the need for `e.preventDefault()` in your click handlers.